### PR TITLE
Small changes in README

### PR DESCRIPTION
**Pull request by zMoooooritz**
Really like the overhauled README looks good!

Here are some (maybe nitpicky) changes to the README.
Furthermore I do not fully understand the keycode action, as I dont know how to get these for a specific keypress.
Therefore there should probably be a link to a resource (that I cant find myself :/) or explanation as to how to determine the keycodes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -198,11 +197,11 @@ You can hook up any key with one of several actions:
   exec = "some_command --with-parameters"
 ```
 
-#### Emulate a key-press
+#### Emulate a (series of) key-press
 
 ```
 [keys.action]
-  keycode = "114"
+  keycode = "9-20+200 / 46"
```
Comment (muesli) [inline README.md:204]: I feel like we should have two examples here, actually: one with a simple keycode, the other explaining serial key-codes and the delays between them.

File: README.md [hunk 2]
```diff
@@ -198,11 +197,11 @@ You can hook up any key with one of several actions:
   exec = "some_command --with-parameters"
 ```
 
-#### Emulate a key-press
+#### Emulate a (series of) key-press
```
Comment (muesli) [inline README.md:200]: Let's keep it simple in the captions: how about just `Emulate key-presses`?

**Comment by muesli** [comment]
> Furthermore I do not fully understand the keycode action, as I dont know how to get these for a specific keypress.
> Therefore there should probably be a link to a resource (that I cant find myself :/) or explanation as to how to determine the keycodes.

Good point! You could look them up in the kernel sources, but while it's not hard and fairly self-explanatory, it's not the most convenient thing either: https://github.com/torvalds/linux/blob/master/include/uapi/linux/input-event-codes.h

**Comment by muesli** [comment]
The Arch wiki also has a few words about identifying keycodes: https://wiki.archlinux.org/title/Keyboard_input

TLDR: it's complicated :laughing:

**Comment by zMoooooritz** [comment]
> Good point! You could look them up in the kernel sources, but while it's not hard and fairly self-explanatory, it's not the most convenient thing either: https://github.com/torvalds/linux/blob/master/include/uapi/linux/input-event-codes.h

Ok, found my mistake. I thought these keycodes where specific to go. The result of that was, that I landed in my search for example here: [https://github.com/yakshaveinc/go-keycodes/blob/master/keycodes.go](https://github.com/yakshaveinc/go-keycodes/blob/master/keycodes.go)

Regarding your comment about the arch wiki entry. This does indeed make it more complicated as there are too many "wrong" keycodes.

Maybe I was just a bit too stupid to come to the conclusion that these have to be the linux keycodes. But nevertheless I personally would add the link to `.../input-event-codes.h`

**Comment by muesli** [comment]
> Ok, found my mistake. I thought these keycodes where specific to go. The result of that was, that I landed in my search for example here: https://github.com/yakshaveinc/go-keycodes/blob/master/keycodes.go

Oh I see! At a first glance this looks like the raw USB scancodes of a keyboard.

> Regarding your comment about the arch wiki entry. This does indeed make it more complicated as there are too many "wrong" keycodes.

Not necessarily wrong, but yeah... how to retrieve them depends heavily on your system: are you in a virtual console? Running Xorg? Running wayland?

> Maybe I was just a bit too stupid to come to the conclusion that these have to be the linux keycodes. But nevertheless I personally would add the link to `.../input-event-codes.h`

Maybe I found something better, since I just remembered working on these uinput bindings for Go: https://github.com/bendahl/uinput/blob/master/keycodes.go

This is a fairly readable list without all the other constants defined in `input-event-codes.h` :smiley:

**Comment by zMoooooritz** [comment]
> Not necessarily wrong, but yeah... how to retrieve them depends heavily on your system: are you in a virtual console? Running Xorg? Running wayland?

I thind no one wants to dive that deep into keycodes just to configure a streamdeck :smile: 

> Maybe I found something better, since I just remembered working on these uinput bindings for Go: https://github.com/bendahl/uinput/blob/master/keycodes.go
> 
> This is a fairly readable list without all the other constants defined in `input-event-codes.h`

Yes this is exactly what I would like to find there!
Maybe we could also include this file and add some code to allow for: `KeyLeftctrl-KeyV+200`
This would results in (series of) keycodes that are not completly cryptic :laughing:

**Comment by muesli** [comment]
> Yes this is exactly what I would like to find there!
> Maybe we could also include this file and add some code to allow for: `KeyLeftctrl-KeyV+200`
> This would results in (series of) keycodes that are not completly cryptic laughing

Agreed, that would be nice! I'd like to think I even worked on something like that before. Possibly in the `uinput` repository as well. Let me check after lunch :smile:

**Comment by muesli** [comment]
Once this PR is merged, I think I'll release `v0.4.0` and push new binaries/packages.

**Comment by zMoooooritz** [comment]
> Once this PR is merged, I think I'll release `v0.4.0` and push new binaries/packages.

Alright, just out of interest. What do you have to do to release / push a new set of binaries/packages. Is this automated and directly linked to the aur and so on?
Or is it just about creating a new tag more or less?

**Comment by muesli** [comment]
> Alright, just out of interest. What do you have to do to release / push a new set of binaries/packages. Is this automated and directly linked to the aur and so on?
> Or is it just about creating a new tag more or less?

It's fairly automated. You can find a `.goreleaser.yml` config in this repo. With this in place, I manually `git tag` a new release and then simply run `goreleaser release`. This builds all the packages for all platforms & architectures, drafts a new release on GitHub, creates a changelog from the new commits, and uploads all the binaries.

Bumping the version number in the AUR is the last manual step in this entire process.

**Comment by muesli** [comment]
Because I just stumbled upon it, you can actually see an example of goreleaser running a test build for every commit: https://github.com/muesli/deckmaster/runs/2741407227
